temporal-sdk/src/test/java/io/temporal/workflow/updateTest/UpdateWithStartTest.java

```diff
@@ -0,0 +1,313 @@
+        WorkflowStartOperationUpdate.newBuilder(
+                TestWorkflows.WorkflowWithUpdate::update, 0, "Hello Update")
+            .build();
+    WorkflowOptions options =
+        SDKTestOptions.newWorkflowOptionsWithTimeouts(testWorkflowRule.getTaskQueue()).toBuilder()
+            .setWorkflowId(workflowId)
+            .setStartOperation(update)
+            .build();
+    TestWorkflows.WorkflowWithUpdate workflow =
+        workflowClient.newWorkflowStub(TestWorkflows.WorkflowWithUpdate.class, options);
+
+    try {
+      WorkflowClient.start(workflow::execute);
+      fail("unreachable");
+    } catch (WorkflowServiceException e) {
+      assertEquals(e.getCause().getMessage(), "waitForStage must not be null");
+    }
+
+    ensureNoWorkflowStarted(workflowClient, workflowId);
+  }
+
+  @Test
+  public void failWhenMethodReferenceThatIsNotUpdateMethod() {
+    try {
+      WorkflowStartOperationUpdate.newBuilder(TestWorkflows.WorkflowWithUpdate::execute).build();
+      fail("unreachable");
+    } catch (IllegalArgumentException e) {
+      assertEquals(e.getMessage(), "Method 'execute' does not have UpdateMethod annotation");
+    }
+
+    try {
+      WorkflowStartOperationUpdate.newBuilder(Integer::reverse).build();
+      fail("unreachable");
+    } catch (IllegalArgumentException e) {
+      assertTrue(e.getMessage().contains("Invalid method reference"));
+    }
+  }
+
+  @Test
+  public void failWhenMethodReferenceDoesNotBelongToWorkflow()
+      throws ExecutionException, InterruptedException {
+    String workflowId = UUID.randomUUID().toString();
+    WorkflowClient workflowClient = testWorkflowRule.getWorkflowClient();
+
+    WorkflowStartOperationUpdate<Void> update =
+        WorkflowStartOperationUpdate.newBuilder(
+                TestWorkflows.TestUpdatedWorkflow::update,
+                "Hello Update") // NOTE: different interface!
+            .setWaitForStage(WorkflowUpdateStage.ACCEPTED)
+            .build();
+    WorkflowOptions options =
+        SDKTestOptions.newWorkflowOptionsWithTimeouts(testWorkflowRule.getTaskQueue()).toBuilder()
+            .setWorkflowId(workflowId)
+            .setStartOperation(update)
+            .build();
+    TestWorkflows.WorkflowWithUpdate workflow =
+        workflowClient.newWorkflowStub(TestWorkflows.WorkflowWithUpdate.class, options);
+
+    try {
+      WorkflowClient.start(workflow::execute);
+      fail("unreachable");
+    } catch (WorkflowServiceException e) {
+      assertEquals(
+          e.getCause().getMessage(),
+          "Update Operation does not belong to 'WorkflowWithUpdate' but 'TestUpdatedWorkflow'");
+    }
+  }
+
+  private static void ensureNoWorkflowStarted(WorkflowClient workflowClient, String workflowId) {
+    try {
+      workflowClient.fetchHistory(workflowId);
+      fail("unreachable");
+    } catch (StatusRuntimeException e) {
+      assertEquals(e.getStatus().getCode(), Status.NOT_FOUND.getCode());
+    }
+  }
+
```

Comment:
I wanted to write a test for a failed server-side validation of the Update, but couldn't find any validation in the test server to leverage.